include/git2/rebase.h

```diff
@@ -23,6 +23,21 @@
  */
 GIT_BEGIN_DECL
 
+/**
+ * Rebase commit signature callback.
+ *
+ * The callback will be called with the commit content, giving a user an
+ * opportunity to sign the commit content in a rebase. The signature_field
+ * buf may be left empty to specify the default field.
+ *
+ * When the callback:
+ * - returns GIT_PASSTHROUGH, no signature will be added to the commit.
+ * - returns < 0, git_rebase_commit will be aborted.
+ * - returns GIT_OK, the signature parameter is expected to be filled.
+ */
+typedef int (*git_rebase_commit_signature_cb)(
```

Comment:
I'm not sure it makes sense for this to be in the rebase namespace, though I can live with it. Are there other places where this kind of signing-trampoline would be useful (ie. cherry-pick) ?

But `signature` makes it sound like `git_signature` are involved, so maybe `git_(rebase_)?commit_signing_cb` ?

Reply:
Currently, I don't believe cherrypick, merge, or revert actually create commits. If we were intending on extending those to create commits rather than an index, I would say it could very well be useful.